src/Diagnostics/cfl.jl

```diff
@@ -72,31 +73,13 @@ DiffusiveCFL(Δt) = CFL(Δt, cell_diffusion_timescale)
 ##### Accurate CFL via reduction
 #####
 
-using CUDA, CUDAKernels, KernelAbstractions, Tullio
-
 using Oceananigans.Models
-using Oceananigans.Grids: halo_size
-using Oceananigans.Operators: Δxᶠᶜᶜ, Δyᶜᶠᶜ, Δzᶜᶜᶠ
-
-accurate_cell_advection_timescale(model) = accurate_cell_advection_timescale(model.grid, model.velocities)
-
-function accurate_cell_advection_timescale(grid, velocities)
-    Nx, Ny, Nz = size(grid)
-    Hx, Hy, Hz = halo_size(grid)
-
-    is = 1+Hx:Nx+Hx
-    js = 1+Hy:Ny+Hy
-    ks = 1+Hz:Nz+Hz
 
-    u = view(velocities.u.data.parent, is, js, ks)
-    v = view(velocities.v.data.parent, is, js, ks)
-    w = view(velocities.w.data.parent, is, js, ks)
+function accurate_cell_advection_timescale(model)
+    
+    u, v, w = model.velocities 
 
-    min_timescale = minimum(
-        @tullio (min) timescale[k] := 1 / (  abs(u[i, j, k]) / Δxᶠᶜᶜ(i, j, k, grid)
-                                           + abs(v[i, j, k]) / Δyᶜᶠᶜ(i, j, k, grid)
-                                           + abs(w[i, j, k]) / Δzᶜᶜᶠ(i, j, k, grid))
-    )
+    min_timescale = minimum(1 / (abs(u) / Δx + abs(v) / Δy + abs(w) / Δz))
```

Comment:
```suggestion
    min_timescale = 1 / max(maximum(abs(u) / Δx), maximum(abs(v) / Δy), maximum(abs(w) / Δz))
```

Reply:
I think in this way you don't account for diagonal motion?

Reply:
I guess wikipedia disagrees with me:

https://en.wikipedia.org/wiki/Courant%E2%80%93Friedrichs%E2%80%93Lewy_condition#The_two_and_general_n-dimensional_case

But I'm not sure why. I'll change the tests....

Reply:
I think it has to do with diagonal motion...

Reply:
🤔

Reply:
This operation is doing interpolation under the hood because of the staggered grid. What's the correct CFL condition for a staggered grid?